flake.crane.nix

```diff
@@ -313,7 +313,17 @@ rec {
     pname = "${commonCliTestArgs.pname}-all";
     version = "0.0.1";
     cargoArtifacts = workspaceBuild;
-    buildPhaseCargoCommand = "patchShebangs ./scripts ; ./scripts/test-ci-all.sh";
+    # One normal run, then if succeeded, modify the "always success test" to fail,
+    # and make sure we detect it (happened too many times that we didn't).
+    # Thanks to early termination, this should be all very quick, as we actually
+    # won't start other tests.
+    buildPhaseCargoCommand = ''
+      patchShebangs ./scripts
+      ./scripts/test-ci-all.sh || exit 1
+      sed -i -e 's/exit 0/exit 1/g' scripts/always-success-test.sh
+      echo "Verifing failure detection..."
+      ./scripts/test-ci-all.sh 1>/dev/null 2>/dev/null && exit 1
```

Comment:
Why `&& exit `?

Reply:
I'm not sure if `set -e` is set here.

So if the `./scripts/test-ci-all.sh` fails, exit 1 will not run and the whole thing return success.

If `./scripts/test-ci-all.sh` succeeds, then `exit 1` runs and we fail.

In other words: making sure we failed after error was introduced.